server_environment/models/server_env_tech_name_mixin.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2020 Camptocamp (http://www.camptocamp.com)
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+from odoo.addons.http_routing.models.ir_http import slugify
+
+
+class ServerEnvTechNameMixin(models.AbstractModel):
+    """Provides a tech_name field to be used in server env vars as unique key.
+
+    The `name` field can be error prone because users can easily change it
+    to something more meaningful for them or set weird chars into it
+    that make difficult to reference the record in env var config.
+    This mixin helps solve the problem by providing a tech name field
+    and a cleanup machinery as well as a unique constrain.
+
+    To use this mixin add it to the _inherit attr of your module like:
+
+        _inherit = [
+            "my.model",
+            "server.env.techname.mixin",
+            "server.env.mixin",
+        ]
+
+    """
+
+    _name = "server.env.techname.mixin"
+    _description = "Server environment technical name"
+    _sql_constraints = [
+        ("tech_name_uniq", "unique(tech_name)", "`tech_name` must be unique!",)
+    ]
+
+    tech_name = fields.Char(
+        required=True,
+        help="Unique name for technical purposes. " "Eg: server env keys.",
+    )
```

Comment:
this field seems to be a good candidate for using the new editable and stored computed field, that would remove the need for the onchange, create, write...

Reply:
added to ROADMAP :wink: